go/lang/correctness/use-filepath-join.yaml

```diff
@@ -1,16 +1,78 @@
 rules:
-- id: use-filepath-join
-  languages: [go]
-  severity: WARNING
-  message: >-
-    `path.Join(...)` always joins using a forward slash. This may cause
-    issues on Windows or other systems using a different delimiter. Use
-    `filepath.Join(...)` instead which uses OS-specific path separators.
-  pattern: path.Join(...)
-  metadata:
-    category: correctness
-    references:
-    - https://parsiya.net/blog/2019-03-09-path.join-considered-harmful/
-    - https://go.dev/src/path/path.go?s=4034:4066#L145
-    technology:
-    - go
+  - id: use-filepath-join
+    languages:
+      - go
+    severity: WARNING
+    message: "`path.Join(...)` always joins using a forward slash. This may cause
+      issues on Windows or other systems using a different delimiter. Use
+      `filepath.Join(...)` instead which uses OS-specific path separators."
+    patterns:
+      - pattern: path.Join(...)
```

Comment:
Is there any reason we can't do taint with `($INPUT : string)` as a source and `path.Join(...)` as a sink?

This many conditions just seems like "code smell" for a taint approach

Reply:
@kurt-r2c I think that semgrep shall support semgrep rules language :)

Reply:
Changed to a taint rule, with functions that return path strings in stdlib as sources.